src/devices/Mcp23xxx/Mcp23xxx.cs

```diff
@@ -439,11 +438,48 @@ private byte GetMappedAddress(Register register, Port port = Port.PortA, BankSty
             return port == Port.PortA ? address : address += 0x10;
         }
 
+        public void OpenPin(int pinNumber) => SetPinMode(pinNumber, PinMode.Input);
+
         public void OpenPin(int pinNumber, PinMode mode) => SetPinMode(pinNumber, mode);
 
         public void ClosePin(int pinNumber)
         {
             // No-op
         }
+
+        public bool IsPinOpen(int pinNumber)
+        {
+            ValidatePin(pinNumber);
+            return true;
+        }
+
+        public PinMode GetPinMode(int pinNumber)
+        {
+            ValidatePin(pinNumber);
+
+            byte direction;
+            if (pinNumber < 9)
+            {
+                direction = InternalReadByte(Register.IODIR, Port.PortA);
+            }
+            else
+            {
+                direction = InternalReadByte(Register.IODIR, Port.PortB);
+            }
+
+            return ((direction >> pinNumber - 1) & 1) == 1 ? PinMode.Input : PinMode.Output;
```

Comment:
Can we add parenthesis on the `direction >> pinNumber - 1`, just so that we make it obvious and order of operations make sense.